homeassistant/components/apcupsd/binary_sensor.py

```diff
@@ -1,43 +1,82 @@
 """Support for tracking the online status of a UPS."""
 from __future__ import annotations
 
-import voluptuous as vol
+import logging
 
-from homeassistant.components.binary_sensor import PLATFORM_SCHEMA, BinarySensorEntity
-from homeassistant.const import CONF_NAME
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import DOMAIN, KEY_STATUS, VALUE_ONLINE
+from . import DOMAIN, VALUE_ONLINE, APCUPSdData
 
-DEFAULT_NAME = "UPS Online Status"
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string}
-)
+_LOGGER = logging.getLogger(__name__)
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up an APCUPSd Online Status binary sensor."""
-    apcups_data = hass.data[DOMAIN]
+    data_service: APCUPSdData = hass.data[DOMAIN][config_entry.entry_id]
+
+    description = BinarySensorEntityDescription(
```

Comment:
If the description is static we should move it to a constant at the module level.

Reply:
Agreed, fixed!